pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/MainServer.java

```diff
@@ -58,26 +62,32 @@ public static void main(String[] args) {
 
         AnnotatorService pipeline = null;
         try {
-            logger.debug("Starting to load the pipeline . . . ");
+            logger.info("Starting to load the pipeline . . . ");
             printMemoryDetails();
             ResourceManager rm = new PipelineConfigurator().getDefaultConfig();
             pipeline = PipelineFactory.buildPipeline(rm);
-            logger.debug("Done with loading the pipeline  . . .");
+            logger.info("Done with loading the pipeline  . . .");
             printMemoryDetails();
         } catch (IOException | AnnotatorException e) {
             e.printStackTrace();
         }
 
         AnnotatorService finalPipeline = pipeline;
         get("/annotate", "application/json", (request, response)->{
+            logger.info("GET request . . . ");
+            logger.info( "request.body(): " + request.body());
             String text = request.queryParams("text");
             String views = request.queryParams("views");
             return annotateText(finalPipeline, text, views);
         });
 
         post("/annotate", (request, response) -> {
-                String text = request.queryParams("text");
-                String views = request.queryParams("views");
+                logger.info("POST request . . . ");
+                logger.info( "request.body(): " + request.body());
+                Map<String, String> map = splitQuery(request.body());
+                System.out.println("POST body parameters parsed: " + map);
+                String text = map.get("text");
+                String views = map.get("views");
```

Comment:
Looks good to me. Did you try to send a post request to see if it works via Postman or similar tool?